toast-tk-plugin-loader/src/main/java/io/toast/tk/plugin/PluginLoader.java

```diff
@@ -0,0 +1,98 @@
+package io.toast.tk.plugin;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Module;
+
+import io.toast.tk.agent.config.AgentConfigProvider;
+import io.toast.tk.plugin.IAgentPlugin;
+
+public class PluginLoader {
+
+	private static final Logger LOG = LoggerFactory.getLogger(PluginLoader.class);
+	private final String pluginDir;
+
+	public PluginLoader(AgentConfigProvider provider) {
+		this(provider.get().getPluginDir());
+	}
+
+	public PluginLoader(String pluginDir) {
+		this.pluginDir = pluginDir;
+	}
+
+	private static void addSoftwareLibrary(File file,
+			ClassLoader classLoader) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)